extensions/amp-story/1.0/pagination-buttons.js

```diff
@@ -222,50 +208,42 @@ export class PaginationButtons {
   onCurrentPageIndexUpdate_(pageIndex) {
     const totalPages = this.storeService_.get(StateProperty.PAGE_IDS).length;
 
-    if (pageIndex === 0) {
-      this.backButton_.updateState(BackButtonStates.HIDDEN);
-    }
-
-    if (pageIndex > 0) {
-      this.backButton_.updateState(BackButtonStates.PREVIOUS_PAGE);
-    }
+    // Hide back button if no previous page.
+    this.setEnabled_(this.backButton_, pageIndex > 0);
 
     if (pageIndex < totalPages - 1) {
-      this.forwardButton_.updateState(ForwardButtonStates.NEXT_PAGE);
+      this.forwardButton_.updateState(ButtonStates.NEXT_PAGE);
     }
 
     if (pageIndex === totalPages - 1) {
       const viewer = Services.viewerForDoc(this.ampStory_.element);
       if (viewer.hasCapability('swipe')) {
-        this.forwardButton_.updateState(ForwardButtonStates.NEXT_STORY);
+        this.forwardButton_.updateState(ButtonStates.NEXT_STORY);
       } else {
-        this.forwardButton_.updateState(ForwardButtonStates.REPLAY);
+        this.forwardButton_.updateState(ButtonStates.REPLAY);
       }
     }
   }
 
+  /**
+   * Updates element based on visibility state.
+   * @param {!PaginationButton} paginationButton
+   * @param {boolean} isVisible
+   * @private
+   */
+  setEnabled_(paginationButton, isVisible) {
+    const {element} = paginationButton;
+    element.classList.toggle('i-amphtml-story-button-hidden', !isVisible);
+    element.querySelector('button')?.toggleAttribute('disabled', !isVisible);
```

Comment:
Nit: There seems to already be a `buttonElement` variable in the `PaginationButton`, we could use it instead of having extra queries.

Reply:
I had that implemented in a [previous commit](https://github.com/ampproject/amphtml/pull/37098#discussion_r763703913).
We would need to make it a public variable in that case, which [may be confusing](https://github.com/ampproject/amphtml/pull/37098#discussion_r763526498) 🤔 
What do you think?

Reply:
I see... I think the confusing part of this is that the class `PaginationButton` is not a button but a container+button. Maybe we can call them `.containerEl` and `buttonEl`, but ideally (not for now) we could remove the container overall and just have a button.

Or even better, why not implement the `setEnabled` directly in the `PaginationButton` class?